test/e2e/run-test

```diff
@@ -0,0 +1,189 @@
+#!/usr/bin/env bash
+set -euo pipefail
+
+SCRIPTPATH="$(cd "$(dirname "$0")"; pwd -P)"
+MAKEFILE="${SCRIPTPATH}/../../Makefile"
+BUILD_DIR="${SCRIPTPATH}/../../build"
+TEST_FAILURES_LOG="${BUILD_DIR}/test-failures.log"
+
+AEIS="${BUILD_DIR}/ec2-instance-selector"
+
+## Build binary for tests to run 
+make -f $MAKEFILE build
+
+## Print to stderr
+function echoerr() {
+    echo "$@" 1>&2;
+}
+
+## Sort a bash array
+function sort_array() {
+    local input=( "$@" )
+    IFS=$'\n' 
+    local sorted=($(sort <<<"${input[*]}"))
+    unset IFS
+    echo "${sorted[*]}"
+}
+
+## Checks if expected items (consumed as an array in args) 
+## are all contained in the actual list (consumed on stdin)
+function assert_contains_instance_types() {
+    local expected=( "$@" )
+    local actual=()
+    while read actual_input; do
+        actual+=($actual_input)
+    done
+    [[ 0 -eq "${#actual[@]}" ]] && return 1
+    local actual_sorted=($(sort_array "${actual[@]}"))
+    local expected_sorted=($(sort_array "${expected[@]}"))
+    
+    for expectation in "${expected_sorted[@]}"; do
+        if [[ ! " ${actual_sorted[*]} " =~ ${expectation} ]]; then
+            echoerr -e "\t🔺ACTUAL: ${actual_sorted[*]}\n\t🔺Expected: ${expected_sorted[*]}"
+            return 1
+        fi
+    done
+}
+
+## Executes an expected vs actual test execution of amazon-ec2-instance-selector
+## Test success or failure is output to stdout and failures are also logged to a file ($TEST_FAILURES_LOG)
+##        $1        = test name string
+##        shift; $@ = params to amazon-ec2-instance-selector (i.e. --vcpus=2)
+##        STDIN     = expected list
+function execute_test() {
+    local test_name=$1
+    shift
+    local params=( "$@" )
+    local expected=()
+    while read expected_input; do
+        expected+=($expected_input)
+    done
+    [[ 0 -eq "${#expected[@]}" ]] && return 1
+
+    echo "=========================== Test: ${test_name} ==========================="
+
+    for p in "${params[@]}"; do
+        set -x
+        if $AEIS ${p} | assert_contains_instance_types "${expected[*]}"; then  
+            echo "✅ ${test_name} \"$p\" passed!"
+        else
+            echo "❌ Failed ${test_name} \"$p\"" | tee "${TEST_FAILURES_LOG}"
+        fi
+        set +x
+    done   
+    echo -e "========================= End Test: ${test_name} ===========================\n\n" 
+}
+
+## Clean up previous test failures
+rm -f "${TEST_FAILURES_LOG}"
+
+################################################ TESTS ################################################
+
+expected=(t3a.micro t3.micro t2.micro)
+params=(
+    "--memory=2"
+    "--memory=1GiB"
+    "--memory=1 GiB"
+    "--memory=1gb"
+    "--memory=1.0"
+    "--memory-min=1 --memory-max=1"
+    "--memory=1024m"
+)
+test_name="Memory 1 GiB"
```

Comment:
I think it'd help with readability if you:

1. move `test_name` assignment before `expected=( ... )` for each case
2. use `test_name` var for the other tests